src/charm.py

```diff
@@ -56,12 +59,24 @@ def __init__(self, *args):
         self.grafana_source_provider = GrafanaSourceProvider(
```

Comment:
Here we need a similar change as in https://github.com/canonical/prometheus-k8s-operator/pull/253